spring-cql/src/test/java/org/springframework/cassandra/test/integration/EmbeddedCassandraServerHelper.java

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cassandra.test.integration;
+
+import static java.util.concurrent.TimeUnit.*;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.commitlog.CommitLog;
+import org.apache.cassandra.io.util.FileUtils;
+import org.apache.cassandra.service.CassandraDaemon;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.SocketUtils;
+import org.yaml.snakeyaml.reader.UnicodeReader;
+
+/**
+ * Imported Embedded Cassandra server startup helper.
+ *
+ * @author Mark Paluch
+ */
+class EmbeddedCassandraServerHelper {
+
+	private static Logger log = LoggerFactory.getLogger(EmbeddedCassandraServerHelper.class);
+
+	public static final long DEFAULT_STARTUP_TIMEOUT = 10000;
+	public static final String DEFAULT_TMP_DIR = "target/embeddedCassandra";
+	/** Default configuration file. Starts embedded cassandra under the well known ports */
+	private static final String INTERNAL_CASSANDRA_KEYSPACE = "system";
+	private static final String INTERNAL_CASSANDRA_AUTH_KEYSPACE = "system_auth";
+	private static final String INTERNAL_CASSANDRA_TRACES_KEYSPACE = "system_traces";
+
+	private static CassandraDaemon cassandraDaemon = null;
+	private static String launchedYamlFile;
+
+	public static void startEmbeddedCassandra(String yamlFile, long timeout) throws Exception {
+		startEmbeddedCassandra(yamlFile, DEFAULT_TMP_DIR, timeout);
+	}
+
+	public static void startEmbeddedCassandra(String yamlFile, String tmpDir, long timeout) throws Exception {
+		if (cassandraDaemon != null) {
+			/* nothing to do Cassandra is already started */
+			return;
+		}
+
+		if (!StringUtils.startsWith(yamlFile, "/")) {
+			yamlFile = "/" + yamlFile;
+		}
+
+		rmdir(tmpDir);
+		copy(yamlFile, tmpDir);
+		File file = new File(tmpDir + yamlFile);
```

Comment:
Use the `File(parent:String, child:String)` constructor instead and Java will figure out the appropriate file separator.

http://docs.oracle.com/javase/8/docs/api/java/io/File.html#File-java.lang.String-java.lang.String-